randr.go

```diff
@@ -144,6 +170,41 @@ func (m Modes) String() string {
 	return strings.Join(str, " ")
 }
 
+// Generates the monitor id from the edid
+func GenerateMonitorId(edid string) (string, error) {
+	var errEdidCorrupted = errors.New("Edid corrupted: " + edid)
+	if edid[:16] != "00ffffffffffff00" {
+		return "", errEdidCorrupted
+	}
+	edid = edid[16:]
+	edid_bytes, err := hex.DecodeString(edid)
+	if err != nil {
+		return "", err
+	}
+
+	manufacturer_enc := binary.BigEndian.Uint16(edid_bytes[:2])
```

Comment:
Same here: Check that `edid_bytes` has the right (minimum) length.

Reply:
I tested the whole length at the beginning of the method. The EDID needs to be at least 32 hex characters in length (16 for the flag, 4 for the uint16, and 8 for the uint32)